src/components/search/SearchPanel.tsx

```diff
@@ -0,0 +1,233 @@
+      setErrMsg("Give at least one character.");
+      return;
+    }
+    const selectedInSameType = searchResults
+      .filter((e) => e.type == searchMode)
+      .map((e) => e.value);
+    if (selectedInSameType.includes(prefixedRemoved)) {
+      setErrMsg(`You already insert this ${searchMode == "username" ? "host" : "title"} keyword.`);
+      return;
+    }
+    const conditions = [...searchResults, { type: searchMode, value: prefixedRemoved }];
+    setSearchResults(conditions);
+    setSearchMode("title");
+    setSearchText("");
+    updatePosts(conditions);
+  };
+
+  const handleDeleteTag = (index: number) => {
+    const conditions = searchResults.filter((_, i) => i !== index);
+    setSearchResults(conditions);
+    updatePosts(conditions);
+  };
+
+  const getOptions = () => {
+    if (searchText.length == 0 || searchMode != "tag") return [];
+    const options = tagNames.filter((tag) =>
+      tag.toLowerCase().includes(searchText.slice(1).toLowerCase())
+    );
+    const selectedValues = searchResults
+      .filter((result) => result.type === searchMode)
+      .map((result) => result.value);
+    return options.filter((option) => !selectedValues.includes(option)).map((e) => "#" + e);
+  };
+
+  const handleInputValueChange = (value: string) => {
+    setErrMsg("");
+    setSearchText(value);
+    if (value.startsWith("@")) {
+      setSearchMode("username");
+    } else if (value.startsWith("#")) {
+      setSearchMode("tag");
+    } else {
+      setSearchMode("title");
+    }
+  };
+
+  return (
+    <Stack alignItems="center" direction="column">
+      <Autocomplete
+        sx={{ width: "500px" }}
+        disableClearable
+        freeSolo
+        inputValue={searchText}
+        value={searchText}
+        onInputChange={(_, value) => {
+          handleInputValueChange(value);
+        }}
+        onChange={(_, value) => {
+          if (!value) return;
+          handleSubmit(value);
+        }}
+        options={getOptions()}
+        renderInput={(params) => (
+          <TextField
+            {...params}
+            name="search"
+            placeholder="Search: @username, #tag, title"
+            variant="outlined"
+            error={errMsg != ""}
+            InputProps={{
+              ref: params.InputProps.ref,
+              sx: textfield_overwrite,
+              endAdornment: (
+                <Button
+                  sx={{
+                    border: "3px solid black",
+                  }}
+                  disabled={searchMode == "tag"}
+                  variant="contained"
+                  type="submit"
+                  style={submit_layout}
+                  onClick={() => handleSubmit(searchText)}
+                >
+                  <SearchIcon />
+                </Button>
+              ),
+            }}
+          />
+        )}
+      />
+      <Box sx={helperText}>
+        <FormHelperText error>{errMsg != "" && errMsg}</FormHelperText>
+      </Box>
```

Comment:
![image](https://user-images.githubusercontent.com/69895436/228532239-16221466-b652-4f07-812c-116a1f5893bd.png)
Error message should position on the left edge of textfield as design
![image](https://user-images.githubusercontent.com/69895436/228532383-ae08e3e9-9b93-4df6-b110-1e81ec240bcc.png)